developer-hub/src/reference_docs/defensive_programming.rs

```diff
@@ -0,0 +1,558 @@
+//! As our runtime should _never_ panic; this includes carefully handling [`Result`]/[`Option`]
+//! types, eliminating the possibility of integer overflows, converting between number types, or
+//! even handling floating point usage with fixed point arithmetic to mitigate issues that come with
+//! floating point calculations.
+//!
+//! Intentional and predictable design should be our first and foremost
+//! property for ensuring a well running, safely designed system.
```

Comment:
We normally try to do line breaks at exactly 100 chars in the codebase.

Reply:
https://marketplace.visualstudio.com/items?itemName=stkb.rewrap will do the work.

Reply:
Forgot to fmt, my bad, and that extension is super useful!